### PR TITLE
Add FileNamesRegex field to App inspector rules

### PR DESCRIPTION
**Pull request by gfs**

### CODE REVIEW

File: RulesEngine/Rule.cs [hunk 1]
```diff
@@ -38,11 +41,40 @@ public class Rule
...
+            set 
+            { 
+                _updateCompiled = true; 
+                _appliesTo = value; 
+            } 
+        }
+
+        private string[]? _appliesTo;
+
+        [JsonProperty(PropertyName = "file_regexes")]
```
Comment (scovetta) [inline RulesEngine/Rule.cs:60]: Maybe `applies_to_file_regex`?

Comment (gfs) [reply RulesEngine/Rule.cs]: Sure. Changed.

File: RulesEngine/RuleProcessor.cs [hunk 2]
```diff
@@ -216,28 +214,57 @@ void ProcessBoundary(ClauseCapture cap)
...
         /// <returns> List of rules </returns>
-        private IEnumerable<ConvertedOatRule> GetRulesForSingleLanguage(string language)
+        private IEnumerable<ConvertedOatRule> GetRulesByLanguage(string input)
+        {
+            string langid = string.Empty;
+
+            if (EnableCache)
+            {
+                // Make language id for cache purposes
+                langid = string.Join(":", input);
```
Comment (scovetta) [inline RulesEngine/RuleProcessor.cs:232]: I don't think this does anything.

File: RulesEngine/RuleProcessor.cs [hunk 3]
```diff
@@ -216,28 +214,57 @@ void ProcessBoundary(ClauseCapture cap)
...
+        /// <returns> List of rules </returns>
+        private IEnumerable<ConvertedOatRule> GetRulesByFileName(string input)
         {
             string langid = string.Empty;
 
             if (EnableCache)
             {
                 // Make language id for cache purposes
-                langid = string.Join(":", language);
+                langid = string.Join(":", input);
```
Comment (scovetta) [inline RulesEngine/RuleProcessor.cs:261]: Same here, input is just a string.

**Review by scovetta — APPROVED** [review]

File: RulesEngine/RuleProcessor.cs (hunk 2 above)
Comment (gfs) [reply RulesEngine/RuleProcessor.cs:232]:
```suggestion
                langid = string.Join(":", input);
```

This isn't actually new code. But yes, I don't know why that's being appended. I'll remove it.

File: RulesEngine/RuleProcessor.cs (hunk 3 above)
Comment (gfs) [reply RulesEngine/RuleProcessor.cs:261]: Yup. I'll update it.

**Pull request merged**